### PR TITLE
#20820 adding the template save draft

### LINKED ISSUE #20820: Create SaveDraft for templates
Create a rest endpoint for template that saves a draft instead of a new version; with that functionality the auto save will just stored on the top without creating many versions of the template for each save
requirements:
1 - if the template does not have an identifier -> create a new version
2 - if the template does not have an inode -> create a new version 
3 - if the template has an id but does not exists  -> create a new version 
4 - if the working version is the same inode of the live version  -> create a new version 
5 - if the user updating the template is not the same of the user that created the working version  -> create a new version 

Otherwise override the template by using the same inode + id.

### PR DESCRIPTION
**Pull request by jdotcms**
This is the first draft for the template saving as a draft

### CODE REVIEW

**Review by erickgonzalez — APPROVED** [review]
Looks good but need tests

**Review by erickgonzalez — CHANGES_REQUESTED** [review]
I think you changed the wrong postman collection, should be Template one instead of the Theme one.

**Comment by jdotcms** [comment]
> I think you changed the wrong postman collection, should be Template one instead of the Theme one.

Done

**Review by erickgonzalez — APPROVED** [review]

File: dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateAPIImpl.java [hunk 1]
```diff
@@ -466,23 +466,84 @@ public void deleteVersionByInode(final String inode) {
 		Try.run(()->FactoryLocator.getTemplateFactory().deleteTemplateByInode(inode)).onFailure(e -> new RuntimeException(e));
 	}
 
+	/*
```
Comment (nollymar) [inline dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateAPIImpl.java:469]: I would add integration tests to cover this logic. There are some validations here that aren't considered in the postman test included

**Review by nollymar — CHANGES_REQUESTED** [review]

File: dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateAPIImpl.java (hunk 1 above)
Comment (erickgonzalez) [reply dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateAPIImpl.java:469]: done

**Review by nollymar — APPROVED** [review]

**Pull request merged**